crates/subspace-service/src/lib.rs

```diff
@@ -184,6 +187,8 @@ pub enum SubspaceNetworking {
         node: Node,
         /// Bootstrap nodes used (that can be also sent to the farmer over RPC)
         bootstrap_nodes: Vec<Multiaddr>,
+        /// DSN metrics registry (libp2p type).
+        metrics_registry: Arc<RefCell<Option<Libp2pMetricsRegistry>>>,
```

Comment:
Usually when you use `Arc` and `RefCell` in the same context you're doing something wrong. What is going on here?

Reply:
I need to have the `Registry` to be `Clone` but the libp2p version doesn't support it 😞

Reply:
Cloning is actually not necessary, here is a PR: https://github.com/subspace/subspace/pull/1941